homeassistant/components/lutron/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow to configure the Lutron integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+from urllib.error import HTTPError
+
+from pylutron import Lutron
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LutronConfigFlow(ConfigFlow, domain=DOMAIN):
+    """User prompt for Main Repeater configuration information."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """First step in the config flow."""
+
+        # Check if a configuration entry already exists
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
What's the reason for this limitation, it's possible for a home to have multiple lutron controllers.
Couldn't we just deduplicate on the guid?

Reply:
Currently every object is stored in `hass.data[LUTRON_*]`. I want to move this to `hass.data[DOMAIN][entry_id]` in the future. But that would make this PR huge. So for now I propose to keep the limit to 1 so we can remove this limitation once we moved everything to the entry id

Reply:
The way the Lutron system is built, a home may have multiple repeaters but only a single instance is ever used for the telnet connection.  The case of multiple instances would mean multiple homes.  Supporting this case would be an extreme edge case in my mind.  I am a certified Lutron installer as well.